sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -23,9 +25,40 @@
  * This represents proxy configuration to be used in http clients..
  */
 public class ProxyOptions {
-    private static final ClientLogger LOGGER = new ClientLogger(ProxyOptions.class);
+    /**
+     * Represents a list of hosts that should be reached directly, bypassing the proxy.
+     * This is a list of patterns separated by '|'. The patterns may start or end with a '*' for wildcards.
+     * Any host matching one of these patterns will be reached through a direct connection instead of through a proxy.
+     *
+     * Default value is {@code null}
+     */
+    public static final String HTTP_PROXY_NON_PROXY_HOSTS = "http.proxy.non-proxy-hosts";
```

Comment:
We should consider adding a checkstyle rule for property names to follow this naming convention with hyphens for single-term multi-word cases (not as part of this PR but a GitHub issue to add later).

Reply:
Great point! created https://github.com/Azure/azure-sdk-for-java/issues/28094